moto/elasticbeanstalk/models.py

```diff
@@ -148,5 +158,18 @@ def _find_environment_by_arn(self, arn: str) -> FakeEnvironment:
                     return env
         raise KeyError()
 
+    def delete_application(
+        self, application_name: str, terminate_env_by_force: bool
+    ) -> FakeApplication:
+        if application_name:
+            if application_name in self.applications:
+                application = self.applications[application_name]
```

Comment:
I've just tested this out against AWS - they just completely remove the application on deletion. So let's do the same:
`self.applications.pop(application_name)`

The response does not contain any information, at all, so we also don't have to return anything in this method.

Reply:
Thanks for the validation! Also, it's the first time that I have to handle a function that doesn't return anything in Moto, I made the following adjustment to the `delete_application` method in the model, is it compliant with the in-place structure for these scenarios: 

model.py
```python
    def delete_application(
        self,
        application_name: str,
    ) -> None:
        if application_name:
            if application_name in self.applications:
                self.applications.pop(application_name)
            else:
                raise ApplicationNotFound(application_name)
```
responses.py
```python
    def delete_application(self) -> str:
        application_name = self._get_param("ApplicationName")
        self.elasticbeanstalk_backend.delete_application(
            application_name=application_name,
        )
        return DELETE_APPLICATION_TEMPLATE
```

Also, I removed the `TerminateEnvByForce` parameter from the model and response as it doesn't have any impact in this scenario as nothing is returned.